### PR TITLE
WINDUPRULE-483,492,498,499,500,509,484,606,429

### PR DESCRIPTION
**Pull request by mmelko**
https://issues.redhat.com/browse/WINDUPRULE-483
https://issues.redhat.com/browse/WINDUPRULE-484
https://issues.redhat.com/browse/WINDUPRULE-492
https://issues.redhat.com/browse/WINDUPRULE-498
https://issues.redhat.com/browse/WINDUPRULE-499
https://issues.redhat.com/browse/WINDUPRULE-500
https://issues.redhat.com/browse/WINDUPRULE-509
https://issues.redhat.com/browse/WINDUPRULE-606
https://issues.redhat.com/browse/WINDUPRULE-429

You can test it by running ` mvn -DrunTestsMatching=component-changes clean test`

### CODE REVIEW

**Comment by mmelko** [comment]
Depends on #443

**Comment by mmelko** [comment]
retest

**Comment by mmelko** [comment]
retest

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        .when(Or.any(JavaClass.references("org.apache.camel.component.hl7.HL7.terser").at(TypeReferenceLocation.IMPORT).as("terser"),
+                FileContent.from("terser").matches("{*}terser({*}"))
+        )
+        .perform(Iteration.over("default").perform(
+                createHint("Terser language: language renamed",
+                        "`terser` language renamed to `hl7terser`",
+                        "languages", true)
+                        .withQuickfix(createReplaceQuickfix("org.apache.camel.component.hl7.HL7.terser",
+                                "org.apache.camel.component.hl7.HL7.hl17terser"))))
+        .withId("component-changes-00008")
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:193]: This ID is already used above, should it be another one ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}simple({*}property{*})").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("Simple language: property function has been removed",
+                "`property` function has been removed from simple language. Use `exchangeProperty` instead.",
+                "languages", true))
+        .withId("component-changes-00008")
+
+        .addRule()
+        .when(Or.any(JavaClass.references("org.apache.camel.component.hl7.HL7.terser").at(TypeReferenceLocation.IMPORT).as("terser"),
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:184]: Looks weird. 
The second condition works with the first one results, so I dont see the need for the second condition as if the first one passes, global result will be True no matter the result of the second condition.

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+        .addRule()
+        .when(Or.any(
+                XmlFile.matchesXpath("//*/c:from[@uri=windup:matches(self::node(), '{*}consumer.{*}')]")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:from[@uri=windup:matches(self::node(), '{*}consumer.{*}')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}from({*}consumer.{*})").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("Consumer endpoints: options with consumer. prefix have been removed. ",
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:228]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        .when(Or.any(
+                XmlFile.matchesXpath("//*[c:secureXML and (count(c:secureXML/@passPhrase)+count(c:secureXML/@passPhraseByte))<1]/c:secureXML/")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*[b:secureXML and (count(b:secureXML/@passPhrase)+count(b:secureXML/@passPhraseByte))<1]/b:secureXML/")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}secureXML").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("XMLsecure dataformat: The default encryption key has been removed",
+                "The default encryption key has been removed, so it is now mandatory to supply " +
+                        "the key String/bytes if you are using symmetric encryption.",
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:216]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+                                "org.apache.camel.component.hl7.HL7.hl17terser"))))
+        .withId("component-changes-00008")
+
+        .addRule()
+        .when(Or.any(JavaClass.references("org.apache.camel.converter.crypto.CryptoDataFormat{*}").at(TypeReferenceLocation.VARIABLE_DECLARATION),
+                XmlFile.matchesXpath("//*[c:crypto and not(b:crypto/@algorithm)]/c:crypto").namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*[b:crypto and not(b:crypto/@algorithm)]/b:crypto").namespace("b", "http://camel.apache.org/schema/blueprint"))
+        )
+        .perform(createHint("Crypto dataformat: The default encryption algorithm changed to null",
+                "The default encryption algorithm is mandatory changed from `DES/CBC/PKCS5Padding` to null. " +
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:201]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+                "languages", true))
+        .withId("component-changes-00008")
+
+        .addRule()
+        .when(Or.any(JavaClass.references("org.apache.camel.component.hl7.HL7.terser").at(TypeReferenceLocation.IMPORT).as("terser"),
+                FileContent.from("terser").matches("{*}terser({*}"))
+        )
+        .perform(Iteration.over("default").perform(
+                createHint("Terser language: language renamed",
+                        "`terser` language renamed to `hl7terser`",
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:189]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+        .addRule()
+        .when(Or.any(
+                XmlFile.matchesXpath("//*/c:simple[text()=windup:matches(self::node(), '{*}property{*}')]")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:simple[text()=windup:matches(self::node(), '{*}property{*}')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}simple({*}property{*})").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("Simple language: property function has been removed",
+                "`property` function has been removed from simple language. Use `exchangeProperty` instead.",
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:179]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+                }
+            }
+        })
+        .withId("component-changes-00002")
+
+        .addRule()
+        .when(FileContent.matches("{*}.{method}{*}").inFileNamed("{*}.java"))
+        .perform(createHint("Mock: {method} has been removed",
+                "`{method}` has been removed from the mock component's assertion api.",
+                "out_message_removed_from_simple_language_and_mock_component", true))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:138]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 9]
```diff
@@ -0,0 +1,308 @@
...
+        .where("method").matches("(outBody|outHeaders)")
+        .withId("component-changes-00003")
+
+        .addRule()
+        .when(JavaClass.references("org.apache.camel.OutHeaders").at(TypeReferenceLocation.ANNOTATION, TypeReferenceLocation.IMPORT))
+
+        .perform(createHint("@OutHeaders annotation has been removed",
+                "`@OutHeaders` annotation has been removed. Use `@Headers` instead",
+                "out_message_removed_from_simple_language_and_mock_component", true))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:147]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 10]
```diff
@@ -0,0 +1,308 @@
...
+        .withId("component-changes-00004")
+
+        .addRule()
+        .when(XmlFile.matchesXpath("//*/c:simple[text()=windup:matches(self::node(), '{*}out.{params}{*}')]")
+                .namespace("c", "http://camel.apache.org/schema/spring"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:153]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 11]
```diff
@@ -0,0 +1,308 @@
...
+        .where("params").matches("(body|header)")
+        .withId("component-changes-00005")
+
+        .addRule()
+        .when(XmlFile.matchesXpath("//*/b:simple[text()=windup:matches(self::node(), '{*}out.{params}{*}')]")
+                .namespace("b", "http://camel.apache.org/schema/blueprint"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:160]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 12]
```diff
@@ -0,0 +1,308 @@
...
+        .where("params").matches("(body|header)")
+        .withId("component-changes-00006")
+
+        .addRule()
+        .when(FileContent.matches("{*}simple({*}out.{params}{*})").inFileNamed("{*}.java"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:166]: No documentation link here ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 13]
```diff
@@ -0,0 +1,308 @@
...
+boolean has(String input) {
+    if (input.contains("="))
+        if (input.substring(input.indexOf("("), input.lastIndexOf(")")).split(",").length == 1)
+            return true
+    return false
+}
+
+Hint createSimpleLanguageRemovedHint(String removed) {
+    return createHint("Simple language: out.{params} has been removed",
+            "`$removed` has been removed from simple language",
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:79]: No effort on these Hints ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 14]
```diff
@@ -0,0 +1,308 @@
...
+                XmlFile.matchesXpath("//*/c:from[@uri=windup:matches(self::node(), '{*}consumer.{*}')]")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:from[@uri=windup:matches(self::node(), '{*}consumer.{*}')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}from({*}consumer.{*})").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("Consumer endpoints: options with consumer. prefix have been removed. ",
+                "Consumer.options with `consumer.` prefix have been removed. Use options without the prefix" +
+                        "i.e `delay` instead of `consumer.delay`",
+                "using_endpoint_options_with_consumer_prefix", true))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:231]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 15]
```diff
@@ -0,0 +1,308 @@
...
+                XmlFile.matchesXpath("//*[c:secureXML and (count(c:secureXML/@passPhrase)+count(c:secureXML/@passPhraseByte))<1]/c:secureXML/")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*[b:secureXML and (count(b:secureXML/@passPhrase)+count(b:secureXML/@passPhraseByte))<1]/b:secureXML/")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}secureXML").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("XMLsecure dataformat: The default encryption key has been removed",
+                "The default encryption key has been removed, so it is now mandatory to supply " +
+                        "the key String/bytes if you are using symmetric encryption.",
+                "xml_security_dataformat", false))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:217]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 16]
```diff
@@ -0,0 +1,308 @@
...
+
+        .addRule()
+        .when(Or.any(JavaClass.references("org.apache.camel.converter.crypto.CryptoDataFormat{*}").at(TypeReferenceLocation.VARIABLE_DECLARATION),
+                XmlFile.matchesXpath("//*[c:crypto and not(b:crypto/@algorithm)]/c:crypto").namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*[b:crypto and not(b:crypto/@algorithm)]/b:crypto").namespace("b", "http://camel.apache.org/schema/blueprint"))
+        )
+        .perform(createHint("Crypto dataformat: The default encryption algorithm changed to null",
+                "The default encryption algorithm is mandatory changed from `DES/CBC/PKCS5Padding` to null. " +
+                        "Therefore if algorithm hasn't been set yet, it's required to set a value for it",
+                "crypto_dataformat", false))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:203]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 17]
```diff
@@ -0,0 +1,308 @@
...
+        .withId("component-changes-00008")
+
+        .addRule()
+        .when(Or.any(JavaClass.references("org.apache.camel.component.hl7.HL7.terser").at(TypeReferenceLocation.IMPORT).as("terser"),
+                FileContent.from("terser").matches("{*}terser({*}"))
+        )
+        .perform(Iteration.over("default").perform(
+                createHint("Terser language: language renamed",
+                        "`terser` language renamed to `hl7terser`",
+                        "languages", true)
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:190]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 18]
```diff
@@ -0,0 +1,308 @@
...
+        .when(Or.any(
+                XmlFile.matchesXpath("//*/c:simple[text()=windup:matches(self::node(), '{*}property{*}')]")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:simple[text()=windup:matches(self::node(), '{*}property{*}')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches("{*}simple({*}property{*})").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("Simple language: property function has been removed",
+                "`property` function has been removed from simple language. Use `exchangeProperty` instead.",
+                "languages", true))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:180]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 19]
```diff
@@ -0,0 +1,308 @@
...
+        .when(XmlFile.matchesXpath("//*/b:simple[text()=windup:matches(self::node(), '{*}out.{params}{*}')]")
+                .namespace("b", "http://camel.apache.org/schema/blueprint"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
+        .where("params").matches("(body|header)")
+        .withId("component-changes-00006")
+
+        .addRule()
+        .when(FileContent.matches("{*}simple({*}out.{params}{*})").inFileNamed("{*}.java"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
+        .where("params").matches("(body|header)")
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:167]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 20]
```diff
@@ -0,0 +1,308 @@
...
+                .namespace("c", "http://camel.apache.org/schema/spring"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
+        .where("params").matches("(body|header)")
+        .withId("component-changes-00005")
+
+        .addRule()
+        .when(XmlFile.matchesXpath("//*/b:simple[text()=windup:matches(self::node(), '{*}out.{params}{*}')]")
+                .namespace("b", "http://camel.apache.org/schema/blueprint"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
+        .where("params").matches("(body|header)")
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:161]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 21]
```diff
@@ -0,0 +1,308 @@
...
+        .perform(createHint("@OutHeaders annotation has been removed",
+                "`@OutHeaders` annotation has been removed. Use `@Headers` instead",
+                "out_message_removed_from_simple_language_and_mock_component", true))
+        .withId("component-changes-00004")
+
+        .addRule()
+        .when(XmlFile.matchesXpath("//*/c:simple[text()=windup:matches(self::node(), '{*}out.{params}{*}')]")
+                .namespace("c", "http://camel.apache.org/schema/spring"))
+        .perform(createSimpleLanguageRemovedHint("out.{params}"))
+        .where("params").matches("(body|header)")
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:154]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 22]
```diff
@@ -0,0 +1,308 @@
...
+        .where("method").matches("(outBody|outHeaders)")
+        .withId("component-changes-00003")
+
+        .addRule()
+        .when(JavaClass.references("org.apache.camel.OutHeaders").at(TypeReferenceLocation.ANNOTATION, TypeReferenceLocation.IMPORT))
+
+        .perform(createHint("@OutHeaders annotation has been removed",
+                "`@OutHeaders` annotation has been removed. Use `@Headers` instead",
+                "out_message_removed_from_simple_language_and_mock_component", true))
+        .withId("component-changes-00004")
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:148]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 23]
```diff
@@ -0,0 +1,308 @@
...
+            }
+        })
+        .withId("component-changes-00002")
+
+        .addRule()
+        .when(FileContent.matches("{*}.{method}{*}").inFileNamed("{*}.java"))
+        .perform(createHint("Mock: {method} has been removed",
+                "`{method}` has been removed from the mock component's assertion api.",
+                "out_message_removed_from_simple_language_and_mock_component", true))
+        .where("method").matches("(outBody|outHeaders)")
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:139]: No effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 24]
```diff
@@ -0,0 +1,308 @@
...
+        .where("regex").matches(telegramRegex)
+        .withId("component-changes-00001")
+
+        .addRule()
+        .when(JavaClass.references("org.apache.camel.component.shiro.security.ShiroSecurityPolicy{*}"))
+        .perform(new AbstractIterationOperation<JavaTypeReferenceModel>() {
+            void perform(GraphRewrite event, EvaluationContext context, JavaTypeReferenceModel payload) {
+
+                Hint h = createHint("Shiro - default encryption key was removed",
+                        "It's mandatory to specify key/passphrase for `ShiroSecurityPolicy`.", "shiro_component", true)
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:123]: No link and no effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 25]
```diff
@@ -0,0 +1,308 @@
...
+                        .matches("$regexReference")
+                        .inFileNamed("{*}.java"),
+                XmlFile.matchesXpath("//*/c:route/*[contains(@uri ,'telegram') and not(contains(@uri,'authorizationToken'))]")
+                        .namespace("c", "http://camel.apache.org/schema/spring")
+        ))
+        .perform(
+                Iteration.over()
+                        .perform(createHint("[camel-telegram] Authorization token parameter required!",
+                                "Authorization token was moved from uri-path to a query parameter. " +
+                                        "Use `telegram:bots?authorizationToken=myTokenHere`",
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:111]: No link nor effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 26]
```diff
@@ -0,0 +1,308 @@
...
+                .with(Link.to("Tracer in Camel 3","https://camel.apache.org/manual/latest/tracer.html"))
+                .withEffort(2))
+        .withId("component-changes-00013")
+
+        .addRule()
+        .when( XmlFile.matchesXpath("/m:project/m:dependencies[m:dependency/m:artifactId/text() = 'camel-core']")
+                .inFile("pom.xml").namespace("m", "http://maven.apache.org/POM/4.0.0"))
+        .perform(createHint("Tracing:  BacklogTracer is no longer enabled by default in JMX",
+                "`BacklogTracer` is no longer enabled by default in JMX. For using BacklogTracer " +
+                        "you need to enable by setting `backlogTracing=true` on CamelContext.",
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:265]: No link nor effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 27]
```diff
@@ -0,0 +1,308 @@
...
+                XmlFile.matchesXpath("//*/c:to[contains(@uri,'xmlsecurity')]")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:to[contains(@uri,'xmlsecurity')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches(".to({*}xmlsecurity{*}").inFileNamed("{*}.java")//
+        )
+        )
+        .perform(createHint("XMLSecurity component: The default signature algorithm has changed",
+                "The default signature algorithm in the XMLSecurity component has changed `SHA1WithDSA` to `SHA256withRSA`. ",
+                "using_endpoint_options_with_consumer_prefix", IssueCategoryRegistry.INFORMATION))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:280]: No link nor effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 28]
```diff
@@ -0,0 +1,308 @@
...
+        .when(Or.any(
+                XmlFile.matchesXpath("//*/c:route/*[contains(@uri,'crypto:')]")
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:route/*[contains(@uri,'crypto:')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches(".to({*}crypto:{*}").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("Crypto component: The default signature algorithm has changed",
+                "The default signature algorithm in the Crypto component has changed from `SHA1WithDSA` to `SHA256withRSA`.",
+                "using_endpoint_options_with_consumer_prefix", IssueCategoryRegistry.INFORMATION))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:293]: No link nor effort ?

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy [hunk 29]
```diff
@@ -0,0 +1,308 @@
...
+                        .namespace("c", "http://camel.apache.org/schema/spring"),
+                XmlFile.matchesXpath("//*/b:route/*[contains(@uri,'xslt:') and contains(@uri,'saxon=true')]")
+                        .namespace("b", "http://camel.apache.org/schema/blueprint"),
+                FileContent.matches(".{*}xslt:{*}saxon=true").inFileNamed("{*}.java"))
+        )
+        .perform(createHint("XSLT: Use xslt-saxon component to use Saxon",
+                "The XSLT component has moved out of `camel-core` into `camel-xslt` and `camel-xslt-saxon`. " +
+                        "Use `xslt-saxon` in URI as described in the documentation.",
+                "xslt", true)
+                .with(Link.to("Camel 3 XSLT Saxon Documentation", "https://camel.apache.org/components/latest/xslt-saxon-component.html")))
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/component-changes.windup.groovy:307]: No effort ?

File: rules-reviewed/camel3/camel2/tests/component-changes.windup.test.xml [hunk 30]
```diff
@@ -0,0 +1,227 @@
...
+                        <iterable-filter size="6">
+                            <hint-exists message="`.*` has been removed from simple language"/>
+                        </iterable-filter>
+                    </not>
+                </when>
+                <perform>
+                    <fail message="[Simple language] outBody/outHeaders removed hint not found!"/>
+                </perform>
+            </rule>
+            <rule id="component-changes-00008-test">
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/tests/component-changes.windup.test.xml:95]: Here there's a test missing for the "Property function has been removed"

File: rules-reviewed/camel3/camel2/tests/component-changes.windup.test.xml [hunk 31]
```diff
@@ -0,0 +1,227 @@
...
+                <when>
+                    <not>
+                        <iterable-filter size="1">
+                            <hint-exists
+                                    message="`org.apache.camel.processor.interceptor.DefaultTraceFormatter` class has been removed.*"/>
+                        </iterable-filter>
+                    </not>
+                </when>
+                <perform>
+                    <fail message="[Tracing] Removed DefaultFormatter class hint not found!"/>
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/tests/component-changes.windup.test.xml:169]: you mean "DefaultTraceFormatter"

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy (hunk 3 above)
Comment (mmelko) [reply rules-reviewed/camel3/camel2/component-changes.windup.groovy:228]: Link is generated in `createHint(....)` method here: https://github.com/windup/windup-rulesets/blob/09f558be161a12b01c91fb5949f98523a0dfc53b/rules-reviewed/camel3/camel2/component-changes.windup.groovy#L50

File: rules-reviewed/camel3/camel2/component-changes.windup.groovy (hunk 2 above)
Comment (mmelko) [reply rules-reviewed/camel3/camel2/component-changes.windup.groovy:184]: that's actually valid point! I believe I wanted to have "And" there. I have to revisit

File: rules-reviewed/camel3/camel2/tests/component-changes.windup.test.xml [hunk 32]
```diff
@@ -16,7 +17,221 @@
...
+                <when>
+                    <not>
+                        <iterable-filter size="2">
+                            <hint-exists
+                                    message="org.apache.camel.processor.interceptor.Tracer` class has been removed.*"/>
+                        </iterable-filter>
+                    </not>
+                </when>
+                <perform>
+                    <fail message="[[Tracing] Removed Tracer class hint not found!"/>
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/tests/component-changes.windup.test.xml:168]: There's an extra [ at the beginning

**Comment by jonathanvila** [comment]
LGTM :+1:

**Pull request merged**